src/streams/mod.rs

```diff
@@ -600,24 +615,23 @@ impl Streams {
     pub(crate) fn change_stream_rx_ssrc(&mut self, ssrc_from: Ssrc, ssrc_to: Ssrc) {
         // This unwrap is OK, because we can't call change_stream_rx_ssrc without first
         // knowing there is such a StreamRx.
-        let mut to_change = self.streams_rx.remove(&ssrc_from).unwrap();
-        to_change.change_ssrc(ssrc_to);
-        let mid = to_change.mid();
-        let rtx = to_change.rtx();
+        let maybe_change = self.streams_rx.get_mut(&ssrc_from).unwrap();
 
-        // Reinsert under new SSRC key.
-        self.streams_rx.insert(ssrc_to, to_change);
+        // The StreamRx is allowed to not change the SSRC in case it is switching back
+        // to the previous SSRC. This is to avoid flapping in case RTP packets arrive
+        // out of order.
+        let did_change = maybe_change.change_ssrc(ssrc_to);
 
-        // Remove previous mappings for the SSRC
-        self.source_keys_rx
-            .retain(|k, (_, s)| *k != ssrc_from && *s != ssrc_from);
+        if did_change {
+            // Unwrap is OK, see above.
+            let to_change = self.streams_rx.remove(&ssrc_from).unwrap();
 
-        // Add mapping for new main SSRC
-        self.associate_ssrc_mid(ssrc_to, mid, ssrc_to, None);
+            // Reinsert under new SSRC key.
+            self.streams_rx.insert(ssrc_to, to_change);
 
-        // Map the RTX to the new main SSRC
-        if let Some(rtx) = rtx {
```

Comment:
We don't need to remap the RTX SSRC anymore?

Reply:
That's the first commit, which means that instead of actively managing a `source_keys_rx`, we are passively using `lookup_rx` as a cache. We can just remove the key from the cache and next time a packet arrives, we will put it back in the cache.

Reply:
Hm, but is that what we want? If we have

```
1 -> (1, A)
2 –> (1, A) (rtx)
```

then change it to

```
3 -> (3, A)
2 -> (1, A) (rtx)
```

if we get a packet on `2` well recreate `1 -> (1, A)`, but don't we want that packet to repair `3`? I guess it doesn't matter since we are repairing the stream `A` after all

Reply:
This would be a change SSRC 1 -> 3, which means the lookup entries for 1 would both be gone since the retain clause is `*k != ssrc_from && l.main != ssrc_from)` – this should clear out both these entires:

```
1 -> (1, A)
2 –> (1, A) (rtx)
```

Reply:
Ah course